### PR TITLE
Add MEGAN for MALT postprocessing

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Adds generation of taxon-table like ouput for MALT (which otherwise only exports a 'custom' file format which is only useful in MEGAN)

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -88,6 +88,7 @@ params {
     // MALT
     run_malt                   = false
     malt_mode                  = 'BlastN'
+    malt_generatemegansummary = false
```
Comment (Midnighter) [inline nextflow.config:91]: Should the default not be `true`? We have agreed to standardize on some kind of tabular format, right?

File: subworkflows/local/profiling.nf [hunk 2]
```diff
@@ -94,30 +96,48 @@ workflow PROFILING {
...
+        ch_maltrun_for_megan = MALT_RUN.out.rma6
+                                .transpose()
+                                .map{
+                                    meta, rma ->
+                                            // re-extract meta from file names, use filename without rma to
+                                            // ensure we keep paired-end information in downstream filenames
+                                            // when no pair-merging
+                                            def meta_new = meta.clone()
+                                            meta_new['db_name'] = meta.id
+                                            meta_new['id'] = rma.name - ( '.' + rma.extension )
```
Comment (Midnighter) [inline subworkflows/local/profiling.nf:109]:
```suggestion
                                            meta_new['id'] = rma.baseName
```

**Review by Midnighter — APPROVED** [review]
Two questions, otherwise this looks good.

File: nextflow.config (hunk 1 above)
Comment (jfy133) [reply nextflow.config:91]: No, this is not a tabular file, but just an extra file that can be loaded into MEGAN (only afaik). The txt file is the thing we want, this is the default output of rma2infilo

**Pull request merged**